tools/duplicateFinder.js

```diff
@@ -0,0 +1,216 @@
+    //lines are parallel
+    let det = A1 * B2 - A2 * B1;
+    if ( det == 0)
+    {
+        //console.log("parrallel")
+        return null; //parallel lines
+    }
+    else
+    {
+        let x = (B2 * C1 - B1 * C2) / det;
+        let y = (A1 * C2 - A2 * C1) / det;
+        //console.log("not parallel", x, y)
+
+        let online1 = ((Math.min(l1p1[0], l1p2[0]) < x || approxeq(Math.min(l1p1[0], l1p2[0]), x))
+            && (Math.max(l1p1[0], l1p2[0]) > x || approxeq(Math.max(l1p1[0], l1p2[0]), x))
+            && (Math.min(l1p1[1], l1p2[1]) < y || approxeq(Math.min(l1p1[1], l1p2[1]), y))
+            && (Math.max(l1p1[1], l1p2[1]) > y || approxeq(Math.max(l1p1[1], l1p2[1]), y))
+            );
+        let online2 = ((Math.min(l2p1[0], l2p2[0]) < x || approxeq(Math.min(l2p1[0], l2p2[0]), x))
+            && (Math.max(l2p1[0], l2p2[0]) > x || approxeq(Math.max(l2p1[0], l2p2[0]), x))
+            && (Math.min(l2p1[1], l2p2[1]) < y || approxeq(Math.min(l2p1[1], l2p2[1]), y))
+            && (Math.max(l2p1[1], l2p2[1]) > y || approxeq(Math.max(l2p1[1], l2p2[1]), y))
+            );
+        if (online1 && online2)
+            return [x, y];
+    }
+    return null; //intersection is at out of at least one segment.
+}
+
+function approxeq  (v1, v2, epsilon = 0.0000001) {
+    return Math.abs(v1 - v2) < epsilon;
+};
+
+function polygonAdd(polygon, point) {
+    //console.log("addpoint", point)
+    let sameElement = polygon.find(( opoint ) => approxeq( point[0], opoint[0]) && approxeq( point[1], opoint[1]));
+    if( typeof sameElement === "undefined") {
+        polygon.push(point)
+    }
+}
+
+function orderClockwise( points ) {
+    let mX = 0;
+    let my = 0;
+    for (let p; p<points.length; p++)
+    {
+        mX += points[p][0];
+        my += points[p][1];
+    }
+    mX /= points.length;
+    my /= points.length;
+    return points.sort(v => Math.atan2(v[1] - my, v[0] - mX));
+}
+
+function polygon_area(vertices){
+    let psum = 0;
+    let nsum = 0;
+
+    for (let i=0; i< vertices.length; i++){
+        sindex = (i + 1) % vertices.length;
+        prod = vertices[i][0] * vertices[sindex][1]
+        psum += prod
+    }
+    for (let i=0; i< vertices.length; i++) {
+        sindex = (i + 1) % vertices.length;
+        prod = vertices[sindex][0] * vertices[i][1]
+        nsum += prod
+    }
+    return Math.abs(1/2*(psum - nsum))
+}
+
+/*
+let tri = [[0,0], [5,5], [0,5]];
+let slab =  [[0, 2.5], [5, 2.5], [5, 5], [0,5]];
+
+console.log(polygon_area(tri));
+console.log(polygon_area(slab)); 
+let isec = intersectPolygons( tri,slab);
+console.log( isec )
+console.log( polygon_area( isec) )
+*/
+/*
+console.log( doBoundsCollide( [ 0, 0, 2, 2 ], [1, 1, 2, 2] )  )
+console.log( doBoundsCollide( [ 0, 0, 2, 2 ], [-1, -1, 2, 2] ))
+console.log( doBoundsCollide( [ 0, 0, 2, 2 ], [2.01, 0, 2, 2] ))
+console.log( doBoundsCollide( [ 0, 0, 2, 2 ], [0, 2.01, 2, 2] ))
+console.log( doBoundsCollide( [ 0, 0, 2, 2 ], [0, -2.01, 2, 2] ))*/
```

Comment:
Include a trailing newline to be [POSIX compliant](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline).